conans/model/conan_file.py

```diff
@@ -183,18 +183,12 @@ def env(self):
     def channel(self):
         if not self._conan_channel:
             self._conan_channel = os.getenv("CONAN_CHANNEL") or self.default_channel
-            if not self._conan_channel:
-                raise ConanException("CONAN_CHANNEL environment variable not defined, "
-                                     "but self.channel is used in conanfile")
```

Comment:
I'm not sure what's the motivation for it right now that we can use references in the `name/version@` form. Is it safe to omit that? It throws an Exception on `inspect.getmembers` otherwise and I'd like to avoid it.

Reply:
Indeed you found a bug! If you try to access `self.channel` in a recipe, which the package has no user/channel, Conan will raise an error.

Reply:
Cool! I'll wrap it into a separate PR once I finish with a PR to the hook repo!